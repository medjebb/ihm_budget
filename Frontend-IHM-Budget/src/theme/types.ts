import type {
  Shadows,
  ColorSystemOptions,
  SupportedColorScheme,
  ThemeOptions as MuiThemeOptions,
} from '@mui/material/styles';

import type { CustomShadows } from './core/custom-shadows';

export type ThemeColorScheme = SupportedColorScheme;
export type ThemeCssVariables = {
  cssVarPrefix?: string;
  colorSchemeSelector?: string;
  disableCssColorScheme?: boolean;
  shouldSkipGeneratingVar?: (keys: string[]) => boolean;
};

type ColorSchemeOptionsExtended = ColorSystemOptions & {
  shadows?: Shadows;
  customShadows?: CustomShadows;
};

export type ThemeOptions = MuiThemeOptions & {
  defaultColorScheme?: ThemeColorScheme;
  colorSchemes?: Partial<Record<ThemeColorScheme, ColorSchemeOptionsExtended>>;
  cssVariables?: ThemeCssVariables;
};
