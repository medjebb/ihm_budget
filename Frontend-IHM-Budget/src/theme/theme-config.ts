import type { CommonColors } from '@mui/material/styles';

import type { ThemeCssVariables } from './types';
import type { PaletteColorNoChannels } from './core/palette';

type ThemeConfig = {
  classesPrefix: string;
};

export const themeConfig: ThemeConfig = {
  classesPrefix: 'min imal',
  // fontFamily: {
  //   primary: 'DM Sans Variable',
  //   secondary: 'Barlow',
  // },
  // palette: {
  // warning: {
  //   lighter: '#FFF5CC',
  //   light: '#FFD666',
  //   main: '#FFAB00',
  //   dark: '#B76E00',
  //   darker: '#7A4100',
  //   contrastText: '#1C252E',
  //   mainChannel: '255, 171, 0',
  //   lightChannel: '255, 214, 102',
  //   darkChannel: '183, 110, 0',
  //   contrastTextChannel: '28, 37, 46',
  // },
  // error: {
  //   lighter: '#FFE9D5',
  //   light: '#FFAC82',
  //   main: '#FF5630',
  //   dark: '#B71D18',
  //   darker: '#7A0916',
  //   contrastText: '#FFFFFF',
  //   mainChannel: '255, 86, 48',
  //   lightChannel: '255, 172, 130',
  //   darkChannel: '183, 29, 24',
  //   contrastTextChannel: '255, 255, 255',
  // },
  // grey: {
  //   '50': '#FCFDFD',
  //   '100': '#F9FAFB',
  //   '200': '#F4F6F8',
  //   '300': '#DFE3E8',
  //   '400': '#C4CDD5',
  //   '500': '#919EAB',
  //   '600': '#637381',
  //   '700': '#454F5B',
  //   '800': '#1C252E',
  //   '900': '#141A21',
  // },
  // common: { black: '#000000', white: '#FFFFFF' },
  // },
  // cssVariables: {
  //   cssVarPrefix: '',
  //   colorSchemeSelector: 'data-color-scheme',
  // },
};
