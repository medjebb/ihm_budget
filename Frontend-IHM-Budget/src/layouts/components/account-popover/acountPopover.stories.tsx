  import type { Meta, StoryObj } from '@storybook/react';
  import HomeIcon from '@mui/icons-material/Home';
  import SettingsIcon from '@mui/icons-material/Settings';
  import PersonIcon from '@mui/icons-material/Person';
  import NotificationsIcon from '@mui/icons-material/Notifications';
  import { MemoryRouter } from 'react-router-dom';
  import { ThemeProvider } from 'src/theme/theme-provider';

  import { AccountPopover } from './account-popover';

  const meta: Meta<typeof AccountPopover> = {
    title: 'Components/AccountPopover',
    component: AccountPopover,
    parameters: {
      layout: 'centered',
    },
    tags: ['autodocs'],
    argTypes: {
      data: {
        control: 'object',
        description: 'Menu items for the popover',
      },
    },
    decorators: [
      (Story) => (
        <ThemeProvider>
          <MemoryRouter>
            <Story />
          </MemoryRouter>
        </ThemeProvider>
      ),
    ],
  };

  export default meta;
  type Story = StoryObj<typeof AccountPopover>;

  export const Default: Story = {
    args: {
      data: [
        {
          label: 'Home',
          href: '/home',
          icon: <HomeIcon />,
        },
        {
          label: 'Profile',
          href: '/profile',
          icon: <PersonIcon />,
        },
        {
          label: 'Settings',
          href: '/settings',
          icon: <SettingsIcon />,
        },
      ],
    },
  };

  export const CustomSize: Story = {
    args: {
      ...Default.args,
      sx: { width: 50, height: 50 },
    },
  };

  export const WithAdditionalInfo: Story = {
    args: {
      data: [
        {
          label: 'Home',
          href: '/home',
          icon: <HomeIcon />,
          info: <span style={{ color: 'green' }}>New</span>,
        },
        {
          label: 'Notifications',
          href: '/notifications',
          icon: <NotificationsIcon />,
          info: <span style={{ color: 'red' }}>2 updates</span>,
        },
      ],
    },
  };

  export const EmptyMenu: Story = {
    args: {
      data: [],
    },
  };

  // export const LongMenuItems: Story = {
  //   args: {
  //     data: [
  //       {
  //         label: 'Very Long Menu Item 1',
  //         href: '/long-item-1',
  //         icon: <HomeIcon />,
  //       },
  //       {
  //         label: 'Another Very Long Menu Item',
  //         href: '/long-item-2',
  //         icon: <SettingsIcon />,
  //       },
  //     ],
  //   },
  // };

  export const DisabledState: Story = {
    args: {
      disabled: true,
    },
  };
