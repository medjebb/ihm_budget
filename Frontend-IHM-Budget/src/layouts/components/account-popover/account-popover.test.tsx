import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'src/theme/theme-provider';
import { AccountPopover } from './account-popover';

jest.mock('src/routes/hooks', () => ({
  useRouter: () => ({ push: jest.fn() }),
  usePathname: () => '/',
}));

const renderComponent = (props = {}) =>
  render(
    <ThemeProvider>
      <MemoryRouter>
        <AccountPopover {...props} />
      </MemoryRouter>
    </ThemeProvider>
  );

describe('AccountPopover', () => {
  it('renders the avatar button', () => {
    const { asFragment } = renderComponent();
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('opens the popover when the avatar button is clicked', () => {
    const { asFragment } = renderComponent();
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(screen.getByRole('presentation')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays menu items when data is provided', () => {
    const data = [
      { label: 'Home', href: '/home', icon: <span>HomeIcon</span> },
      { label: 'Profile', href: '/profile', icon: <span>ProfileIcon</span> },
    ];
    const { asFragment } = renderComponent({ data });

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Profile')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
