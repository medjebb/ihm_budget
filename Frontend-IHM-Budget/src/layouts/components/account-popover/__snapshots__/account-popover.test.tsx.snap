// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPopover displays menu items when data is provided 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-riii7s-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-dlwo3h-MuiAvatar-root"
    >
      <img
        alt="Ahmed Amine Mahamani"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/assets/images/avatar/avatar-25.webp"
      />
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`AccountPopover opens the popover when the avatar button is clicked 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-riii7s-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-dlwo3h-MuiAvatar-root"
    >
      <img
        alt="Ahmed Amine Mahamani"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/assets/images/avatar/avatar-25.webp"
      />
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`AccountPopover renders the avatar button 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-riii7s-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-dlwo3h-MuiAvatar-root"
    >
      <img
        alt="Ahmed Amine Mahamani"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/assets/images/avatar/avatar-25.webp"
      />
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
